### PR TITLE
2.3 Remove team

### PR DESCRIPTION
**Pull request by Y-floryanovich**

### CODE REVIEW

File: src/Teams/Controllers/ManageTeamsController.cs [hunk 1]
```diff
@@ -21,5 +28,10 @@ public IActionResult Error()
         {
             return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
         }
+
+        public void Remove(int team_id)
+        {
+            _teamsService.Remove(team_id);
```
Comment (alehkahnovich) [inline src/Teams/Controllers/ManageTeamsController.cs:34]: await _teamsService.Remove(team_id);

File: src/Teams/Controllers/ManageTeamsController.cs [hunk 2]
```diff
@@ -21,5 +28,10 @@ public IActionResult Error()
         {
             return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
         }
+
+        public void Remove(int team_id)
```
Comment (alehkahnovich) [inline src/Teams/Controllers/ManageTeamsController.cs:32]: Is this is an action method? If so then we should return IActionResult or Task<IActionResult>

File: src/Teams/Services/IManageTeamsService.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace Teams.Services
+{
+    public interface IManageTeamsService
+    {
+        Task<bool> Remove(int team_id);
```
Comment (alehkahnovich) [inline src/Teams/Services/IManageTeamsService.cs:7]: RemoveAsync please, we should stick to naming convention. If the method returns Task then the method name should end with Async

File: src/Teams/Services/ManageTeamsService.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+        public ManageTeamsService(ICurrentUser currentUser, IRepository<Team, int> teamRepository)
+        {
+            _teamRepository = teamRepository;
+            _currentUser = currentUser;
+        }
+
+        public async Task<bool> Remove(int team_id)
+        {
+            var team = _teamRepository.GetAll().FirstOrDefault(i => i.TeamOwner == _currentUser.Current.Id() && i.Id == team_id);
```
Comment (alehkahnovich) [inline src/Teams/Services/ManageTeamsService.cs:25]: There is another version of FirstOrDefaullt extension method, FirstOrDefaulltAsync.

**Review by alehkahnovich — CHANGES_REQUESTED** [review]

File: Teams.Tests/ManageTeamServiceTest.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            private Mock<IHttpContextAccessor> _httpContextAccessor;
+            private Mock<IRepository<Team, int>> _teamRepository;
+            private ICurrentUser _currentUser;
+            private ManageTeamsService _manageTeamsService;
+
+            [SetUp]
+            public void Setup()
+            {
+                _httpContextAccessor = new Mock<IHttpContextAccessor>();
+                _currentUser = new CurrentUser(_httpContextAccessor.Object);
```
Comment (SergeyMokin) [inline Teams.Tests/ManageTeamServiceTest.cs:30]: please use Mock here

File: Teams.Tests/ManageTeamServiceTest.cs [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                _teamRepository = new Mock<IRepository<Team, int>>();
+                _manageTeamsService = new ManageTeamsService(_currentUser, _teamRepository.Object);
+            }
+
+            [Test]
+            public async Task RemoveAsync_ManageTeamsServiceReturnsTrue_ReturnsTrue()
+            {
+                // Arrange
+                string team_owner = "1234";
```
Comment (SergeyMokin) [inline Teams.Tests/ManageTeamServiceTest.cs:39]: can be const

File: Teams.Tests/ManageTeamServiceTest.cs [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                int team_id = 1;
```
Comment (SergeyMokin) [inline Teams.Tests/ManageTeamServiceTest.cs:40]: same here

File: Teams.Tests/ManageTeamServiceTest.cs [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+                var teams = new List<Team>
+            {
+                 new Team { Id= 1, TeamOwner = "1234", TeamName = "First_Team"},
+                 new Team { Id= 2, TeamOwner = "1234", TeamName = "Second_Team"}
+            }.AsQueryable();
+
+                _httpContextAccessor.Setup(x => x.HttpContext.User.FindFirst(It.IsAny<string>()))
+             .Returns(new Claim("UserName", team_owner));
```
Comment (SergeyMokin) [inline Teams.Tests/ManageTeamServiceTest.cs:48]: please take a look into formatting from 41 line to this

File: Teams.Tests/ManageTeamServiceTest.cs [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+                //Act
+                var result = await _manageTeamsService.RemoveAsync(team_id);
+              
+                //Assert
+                Assert.IsTrue(result); 
+            }
+
+            [Test]
+            public async Task RemoveAsync_ManageTeamsServiceReturnsFalse_ReturnsFalse()
```
Comment (SergeyMokin) [inline Teams.Tests/ManageTeamServiceTest.cs:64]: in this test same questions

**Review by SergeyMokin — COMMENTED** [review]
Please resolve conflicts

File: src/Teams/Controllers/ManageTeamsController.cs [hunk 10]
```diff
@@ -21,5 +30,19 @@ public IActionResult Error()
         {
             return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
         }
+
+        [Authorize]
+        public async Task<IActionResult> RemoveAsync(int team_id)
```
Comment (alehkahnovich) [inline src/Teams/Controllers/ManageTeamsController.cs:35]: You can skip Async naming convention on endpoint method level as for the methods in controller you should use default naming convention

**Review by alehkahnovich — CHANGES_REQUESTED** [review]

**Review by alehkahnovich — APPROVED** [review]

**Pull request merged**